### ISSUE TITLE
No buttons text in XMage Launcher when executed form "Program Files" folder

### ISSUE THREAD

**Issue by majaldm** [magefree/mage#4126]
I'm trying to launch the XMage Launcher from Windows' Program Files folder (both C:\Program Files (x86)\XMage and C:\Program Files\XMage) and picture realated is what I get. Launching from any other location yields a normal view of the buttons.

![xmage_nobuttonbs](https://user-images.githubusercontent.com/19721326/31860906-e8450474-b754-11e7-868a-c8c4702dbbd1.jpg)

I have jre1.8.0_151 installed and I'm running Windows 10 64-bit.

**Comment by JayDi85** [comment]
XMage need access to write data files in application directory. There are can be some checks and warnings for users without access. 

Example behaviour:
 - check write access to app dir (sample file);
 - if can't write -- inform user to run app in admin mode);

**Comment by ldeluigi** [comment]
Same problem. The only convenient solution I've found is to place xmage inside a permission-rules free directory.
otherwise you always need to run it as administrator.

**Comment by JayDi85** [comment]
Yes, xmage require write access to running folder. Read only folders support will be added in the feature (see #7025).

**Comment by ldeluigi** [comment]
> Yes, xmage require write access to running folder. Read only folders support will be added in the feature (see #7025, it also contains workaround to run it from read only folder).

Ok but please don't default to AppData (it's a bad idea to download 10GB of images inside it), just ask for a folder where it should download everything and then point to that folder from some form of link put in appdata.

This way you don't touch the jar folder or the environment variables/registry
